src/main/java/com/simpligility/maven/plugins/android/phase09package/ApkMojo.java

```diff
@@ -825,7 +825,7 @@ private File removeDuplicatesFromJar( File in, List<String> duplicates,
                     //if not handled by transformer, add (once) to duplicates jar
                     if ( !resourceTransformed )
                     {
-                        if ( !duplicatesAdded.add( entry.getName() ) )
+                        if ( duplicatesAdded.add( entry.getName() ) )
```

Comment:
Hi William
 
The duplicatesAdded HashSet is modified when the add operation returns true.
Combining a test with an action in one method is efficient but the intention
gets obscured.
The alternative is to do a contains() test and then only proceed with the
add() if the result is false.
 
> @@ -825,7 +825,7 @@ private File removeDuplicatesFromJar( File in,
List<String> duplicates,

>                      //if not handled by transformer, add (once) to
duplicates jar

>                      if ( !resourceTransformed )

>                      {

> -                        if ( !duplicatesAdded.contains( entry.getName() )

                           {
                               duplicatesAdded.add( entry.getName() ) )


Regards
 
Andrew
 

  _____  

From: William Ferguson [mailto:notifications@github.com] 
Sent: Tuesday, 28 July 2015 4:40 PM
To: simpligility/android-maven-plugin
Cc: Andrew Bowley
Subject: Re: [android-maven-plugin] Fix duplicate file detection logic -
Issue 660 (#661)



In
src/main/java/com/simpligility/maven/plugins/android/phase09package/ApkMojo.
java
<https://github.com/simpligility/android-maven-plugin/pull/661#discussion_r3
5617931> :

> @@ -825,7 +825,7 @@ private File removeDuplicatesFromJar( File in,
List<String> duplicates,

>                      //if not handled by transformer, add (once) to
duplicates jar

>                      if ( !resourceTransformed )

>                      {

> -                        if ( !duplicatesAdded.add( entry.getName() ) )

> +                        if ( duplicatesAdded.add( entry.getName() ) )

@andrew-bowley <https://github.com/andrew-bowley>  I think you are correct
in that this code block should only be entered if (duplicatesAdded(.add(
entry.getName() )

But I can't see where duplicatesAdded is ever modified, so it always be an
emptySet.
@kedzie <https://github.com/kedzie>  can you shed some light as this was
code you added in version:6a1a483.

-
Reply to this email directly or view
<https://github.com/simpligility/android-maven-plugin/pull/661/files#r356179
31> it on GitHub.
<https://github.com/notifications/beacon/AIVw2I2GKi36lx8t379vuUDu8eHW1PArks5
ohxsxgaJpZM4Fg3R3.gif>